font/src/directwrite/mod.rs

```diff
@@ -107,73 +231,29 @@ impl crate::Rasterize for DirectWriteRasterizer {
 
                     idx += 1;
                 }
-            },
+            }
         }?;
 
-        let face = font.create_font_face();
-        self.fonts.push(face);
+        let font_face = font.create_font_face();
+        let key = FontKey::next();
+        self.keys.insert(desc.clone(), key);
+        self.fonts.insert(key, LoadedFont { font, font_face });
 
-        Ok(FontKey { token: (self.fonts.len() - 1) as u16 })
+        Ok(key)
     }
 
     fn get_glyph(&mut self, glyph: GlyphKey) -> Result<RasterizedGlyph, Error> {
-        let font = self.fonts.get(glyph.font_key.token as usize).ok_or(Error::FontNotLoaded)?;
-
-        let offset = GlyphOffset { advanceOffset: 0.0, ascenderOffset: 0.0 };
-
-        let glyph_index = *font
-            .get_glyph_indices(&[glyph.c as u32])
-            .first()
-            .ok_or_else(|| Error::MissingGlyph(glyph.c))?;
-        if glyph_index == 0 {
-            // The DirectWrite documentation states that we should get 0 returned if the glyph
-            // does not exist in the font
-            return Err(Error::MissingGlyph(glyph.c));
+        let loaded_font = self.get_loaded_font(glyph.font_key)?;
+
+        match self.rasterize_glyph(&loaded_font.font_face, glyph.c, glyph.size) {
+            Ok(glyph) => Ok(glyph),
+            Err(err @ Error::MissingGlyph(_)) => {
+                let fallback_font =
+                    self.get_fallback_font(&loaded_font.font, glyph.c).ok_or(err)?;
+                self.rasterize_glyph(&fallback_font.create_font_face(), glyph.c, glyph.size)
```

Comment:
This is also super minor, but it seems unfortunate that we *just barely* linebreak here.

I'd be fine with renaming this to `fallback` or just `font`. Though `fallback_font` is definitely the best variable name, so I don't mind that too much either.